### PR TITLE
SDK-2354 added error details and related tests

### PR DESCRIPTION
**Pull request by mehmet-yoti**

### CODE REVIEW

File: src/Yoti.Auth/DigitalIdentity/ErrorDetails.cs
Comment (saurabh-yoti) [inline src/Yoti.Auth/DigitalIdentity/ErrorDetails.cs:1]: `GetErrorReason()` method is not required.

File: src/Yoti.Auth/DigitalIdentity/ErrorReason.cs
Comment (saurabh-yoti) [inline src/Yoti.Auth/DigitalIdentity/ErrorReason.cs:1]: `GetRequirementNotMetDetails()` method is not required.

File: src/Yoti.Auth/DigitalIdentity/RequirementNotMetDetails.cs
Comment (saurabh-yoti) [inline src/Yoti.Auth/DigitalIdentity/RequirementNotMetDetails.cs:1]: Separate Get methods like `GetFailureType()` and `GetDetails()` are not required.

File: test/Yoti.Auth.Tests/DigitalIdentity/RequirementNotMetDetails.cs
Comment (saurabh-yoti) [inline test/Yoti.Auth.Tests/DigitalIdentity/RequirementNotMetDetails.cs:1]: Please update test after making requested changes to `RequirementNotMetDetails` class.

File: src/Yoti.Auth/DigitalIdentity/ShareSessionRequestBuilder.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿using System.Collections.Generic;
 using Yoti.Auth.DigitalIdentity.Extensions;
 using Yoti.Auth.DigitalIdentity.Policy;
+using Yoti.Auth.ShareUrl.Policy;
```
Comment (saurabh-yoti) [inline src/Yoti.Auth/DigitalIdentity/ShareSessionRequestBuilder.cs:4]: Why was this import added - `using Yoti.Auth.ShareUrl.Policy`? It doesn't seem to have any use here.

**Review by saurabh-yoti — CHANGES_REQUESTED** [review]

File: src/Yoti.Auth/DigitalIdentity/ErrorDetails.cs
Comment (mehmet-yoti) [reply src/Yoti.Auth/DigitalIdentity/ErrorDetails.cs:1]: solved

File: src/Yoti.Auth/DigitalIdentity/ErrorReason.cs
Comment (mehmet-yoti) [reply src/Yoti.Auth/DigitalIdentity/ErrorReason.cs:1]: solved

File: src/Yoti.Auth/DigitalIdentity/RequirementNotMetDetails.cs
Comment (mehmet-yoti) [reply src/Yoti.Auth/DigitalIdentity/RequirementNotMetDetails.cs:1]: solved

File: src/Yoti.Auth/DigitalIdentity/ShareSessionRequestBuilder.cs (hunk 1 above)
Comment (mehmet-yoti) [reply src/Yoti.Auth/DigitalIdentity/ShareSessionRequestBuilder.cs]: removed

File: test/Yoti.Auth.Tests/DigitalIdentity/RequirementNotMetDetails.cs
Comment (mehmet-yoti) [reply test/Yoti.Auth.Tests/DigitalIdentity/RequirementNotMetDetails.cs:1]: updated

**Review by saurabh-yoti — APPROVED** [review]
